deployment-scripts/jobs-deployment/tests/test_deploy_job.py

```diff
@@ -228,6 +234,60 @@ def test_should_start_code_job_with_clean_state_if_not_running_and_no_previous_s
             self.flink_cli_runner.start, "python_flink_params", f"--python /tmp/run-{self.TEST_JOB_NAME}.py"
         )
 
+    @mock_s3
+    def test_should_start_job_with_job_status_running_whole_time(self):
+        # given: an empty S3 bucket for state
+        self._create_s3_bucket_for_flink_data()
+
+        # and: FlinkCliRunner mock
+        self.flink_cli_runner.is_job_running = MagicMock(return_value=False)
+        self.flink_cli_runner.get_job_id = MagicMock(return_value="test_job_id")
+        self.flink_cli_runner.stop_with_savepoint = MagicMock()
+        self.flink_cli_runner._get_job_status = MagicMock(side_effect=["CREATED", "RUNNING", "RUNNING", "FAILING", "RESTARTING"])
+        self.flink_cli_runner.start = MagicMock()
+
+        # and: a SQL job manifest
+        job_manifest = self.a_valid_sql_job_manifest().build().to_yaml()
+        config_file_path = self._write_to_local_file(job_manifest)
+
+        # when
+        try:
+            self._run_job(config_file_path)
+
+        # then
+        except RuntimeError as e:
+            self.assertEqual("Unexpected job state. Recent status FAILING.", str(e))
```

Comment:
`str(e)` I expect that the error has a method to extract the error message.

Reply:
It does not. It is recommended to use str(...) for getting the error message.
https://stackoverflow.com/questions/4308182/getting-the-exception-value-in-python